### ISSUE TITLE
Error during composer create-project: Entrypoint undefined = extract-text-webpack-plugin-output-filename

### ISSUE THREAD

**Issue by EstebanForge** [htmlburger/wpemerge-theme#41]
## Version

  - WP Emerge Starter Theme: 0.15.1
  - WordPress: 5.3.2
  - PHP: 7.3.9
  - Composer: 1.9.2
  - Yarn: 1.21.1
  - macOS: Catalina 10.15.2

## Expected behavior

Composer create project should create a project without errors.

## Actual behavior

Got some errors with the default command (from the quickstart https://docs.wpemerge.com/#/starter-theme/quickstart) for a new project:

https://tppr.me/drKtH

```
[130] ./resources/styles/editor/index.scss 41 bytes [built]
    + 120 hidden modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/admin/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [0] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/admin/index.scss 165 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/editor/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [0] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/editor/index.scss 2.26 KiB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/login/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [0] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/login/index.scss 165 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/theme/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [0] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/theme/index.scss 7.23 KiB {0} [built]
        + 1 hidden module
```

## Steps to reproduce (in case of a bug)

  1. Follow the quickstart guide and create a project https://docs.wpemerge.com/#/starter-theme/quickstart
  2. Error during install.

## Comments

If i continue, the theme installation finish. But when trying to develop with it, "yarn dev" result in the same type of error:

```
[./resources/styles/theme/index.scss] 41 bytes [built]
[jquery] external "jQuery" 42 bytes {theme} [built]
    + 123 hidden modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!node_modules/bootstrap/dist/css/bootstrap.css:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/admin/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [./node_modules/css-loader/index.js?!./node_modules/postcss-loader/lib/index.js?!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob/index.js!./resources/styles/admin/index.scss] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/admin/index.scss 165 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/editor/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [./node_modules/css-loader/index.js?!./node_modules/postcss-loader/lib/index.js?!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob/index.js!./resources/styles/editor/index.scss] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/editor/index.scss 3.39 KiB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/login/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [./node_modules/css-loader/index.js?!./node_modules/postcss-loader/lib/index.js?!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob/index.js!./resources/styles/login/index.scss] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/login/index.scss 165 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js??ref--7-1!node_modules/postcss-loader/lib/index.js??ref--7-2!node_modules/sass-loader/lib/loader.js!node_modules/import-glob/index.js!resources/styles/theme/index.scss:
    Entrypoint undefined = extract-text-webpack-plugin-output-filename
    [./node_modules/css-loader/index.js?!./node_modules/postcss-loader/lib/index.js?!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob/index.js!./resources/styles/theme/index.scss] ./node_modules/css-loader??ref--7-1!./node_modules/postcss-loader/lib??ref--7-2!./node_modules/sass-loader/lib/loader.js!./node_modules/import-glob!./resources/styles/theme/index.scss 11.7 KiB {0} [built]
        + 1 hidden module
```

**Comment by atanas-dev** [comment]
Hi @TCattd ,

Apart from the curious `Entrypoint undefined` messages everything else seems as expected - are you actually missing any styles/scripts after building?

I'll look into the `undefined` message :+1:

**Comment by EstebanForge** [comment]
Styles and Scripts are working as far as i have tested.

But i can't get images in /resources/images/ to show up in the dist folder: https://tppr.me/own37
Manifest won't update: https://tppr.me/FIBLN

First time testing this starter theme. Can't find anything in the docs if i'm doing this right or not, really.

**Comment by TCattd** [comment]
I figured it out. Can't find it in the documentation, but after reading the source, i notice that you need to import every asset by hand in /resources/scripts/theme/index.js

I suppose that let us have those nice aliases https://docs.wpemerge.com/#/starter-theme/assets/overview

But if anyone don't want to include all your images by hand, the ones you drop into /resources/images/, you can you add this to your /resources/scripts/theme/index.js:

```
import '@images/{ico,jpg,jpeg,png,svg}';
```

(Or replace the favicon.ico import line with that one)

This way it will import every ico, jpg, png and svg in that folder to the manifest.json for reference.
After that there's no need to manually import every single image as you drop them into /resources/images/

**Comment by EstebanForge** [comment]
Hmm. Isn't mentioned at https://docs.wpemerge.com/#/starter-theme/utilities/theme-assets?id=themeassetsgetasseturi-asset- but you need to reference/import any image you add to /resources/images/ inside

```
/resources/scripts/theme/index.js
```

Like the favicon.ico example

```
import '@images/logo.jpg';
```

Before you can use them.
Then you can do this:

```
<img src="<?php echo Theme\Assets::getAssetUri( 'images/logo.jpg' ); ?>" alt="Logo" />
```

--- Edit ---

(Again) Unrelated to the main issue (don't want to bother you with another issue): is there any way to automatically import all images from /resources/images/ ?
Without having to reference every one of them one by one.

**Comment by TCattd** [comment]
Unrelated to the main issue (don't want to bother you with another issue): is there any way to automatically import all images from /resources/images/ ?
Without having to reference every one of them one by one.

**Comment by mpdevcl** [comment]
Usually, it is preferable to import only the necessary images. But if for some reason it is needed, you can use this snippet to import all the images at once. In resources/scripts/theme/index.js:

```
function importAllImages(r) {
  return r.keys().map(r);
}
const themeImages = importAllImages(require.context('@images/', false, /\.(png|jpe?g|svg)$/));
```

**Comment by atanas-dev** [comment]
1. The undefined issue has been fixed in 6caf0e685985ce23626feb35ba89b8da5a4ecda8 and will be included in the next release. As it is, it doesn't affect the build process and can be ignored.
2. The documentation has been updated (cache may apply) to state that assets must be imported/referenced in order to be processed (https://docs.wpemerge.com/#/starter-theme/assets/overview?id=importing-assets)

Thank you @TCattd and @mpdevcl for the input!

**Comment by EstebanForge** [comment]
Thanks!
